### DISCUSSION TITLE
Browse Examples on crates.io

### DISCUSSION THREAD

**Discussion by m-vz** [rust-lang/crates.io#6554] [Feature Requests]
Creating a feature request issue led me here. I assume a discussion about it is desired before opening an issue...

I noticed that after `Readme.md`, the usual process of evaluating a crate is browsing the `/examples` folder. That folder is a common convention defined by cargo and found in many crates.

When looking at multiple crates, it would be nice to be able to browse the examples without visiting the repository for each one. Something like an "Examples" tab where the files inside `/examples` are displayed.

Thoughts?

**Comment by Turbo87** [comment]
this would require us to build the infrastructure to display code from within the published crates, similar to what docs.rs does. while this seems useful in the long-term, I unfortunately don't think we currently have the capacity to build this in a scalable way.

**Reply by m-vz** [reply]
I see... So the readme is displayed differently then? (I assumed since the readme is also "code" displayed from the repository, showing a different file would only be a minor change.)

This would understandably be too much effort for a minor convenience. Maybe something to keep in mind for if there are ever any other reasons to implement displaying code on crates.io...

**Reply by Turbo87** [reply]
the README content is currently provided by `cargo` when it sends the publish request to crates.io
